docs/src/main/tut/docs/akkahttp.md

```diff
@@ -3,3 +3,102 @@ layout: docs
+The `complete` directive can take just a message, or just the status code to give back; but more frequently, it takes as an argument
+an expression, or program call, whose result is the _response body_ that is given back to the client.
+
+The goal of `freestyle` integration with Akka HTTP is to allow programmers to pass an expression of type `fs : FreeS[F, A]`,
+for some kind-1 type `F[_]` and some base type `A`, as the argument of a `complete` directive. With respect to this goal,
+the [Akka HTTP docs](http://doc.akka.io/docs/akka-http/10.0.5/java/http/introduction.html#routing-dsl-for-http-servers) say that:
+
+> Transforming request [...] bodies between over-the-wire formats and objects to be used in your application
+> is done separately from the route declarations, in marshallers, which are pulled in implicitly using the “magnet” pattern.
+> This means that you can complete a request with any kind of object a as long as there is an implicit marshaller available in scope.
+
+Thus, the `freestyle` integration for Akka HTTP provides an extension of that _magnet_ pattern, which allow us to generate
+response [marshallers](http://doc.akka.io/docs/akka-http/current/scala/http/common/marshalling.html).
+To be precise, what the integration gives is an `implicit` method that may generate an object of type
+
+```Scala
+marsh: ToEntityMarshaller[  FreeS[ F, A ] ]
+```
+for some types parameters `F[_]` , which is the target of the algebra, and for some base result type `A`.
+Note that the method has to be parametrised both on the `F` and on `A`, to make it as generic as possible.
+This is the method we have:
+
+```Scala
+  implicit def seqToEntityMarshaller[F[_], G[_], A](
+      implicit NT: F ~> G,
+      MonG: Monad[G],
+      gem: ToEntityMarshaller[G[A]]
+  ): ToEntityMarshaller[FreeS[F, A]] =
+    gem.compose((fs: FreeS[F, A]) => fs.exec[G])
+```
+
+To build such an object `marsh`, our method needs to find in scope :
+
+* A _base_ entity marshaller, `base: ToEntityMarshaller[ G[A]]`, for a kind-1 type `G[_]` which is determined by the application,
+  and for the same result type `A`.
+* A way to interpret an expression of type `FreeS[F, A]` into a `G[A]`. This is built by the implicit metods in
+  `freestyle.implicits`, from a natural transformation `F ~> G` and an instance of `cats.Monad[G]`.
+
+In essence, the generated marshaller first _interprets_ the value of type `FreeS[F,A]` into a `G[A]`, and then it passes the generated
+value to the base marshaller. To build an Akka HTTP route using a `FreeS` program, one only needs to bring this method into the implicit context, which can be done using an
+`import freestyle.http.akka._` statement.
+
+##### Small Example
+
+As an example, here is a small API program mixing `freestyle` and Akka HTTP. First, we define a domain class `User`, we write
+a `@free` algebra that returns values of a type `FreeS[F, User]`, and then define an interpreter  for
+
+```Scala
```

Comment:
Should we use `tut:book` instead of `scala` ?
That way it is actually checked that the code compiles.